### ISSUE TITLE
Contextual Binding When() Target/Service

### ISSUE THREAD

**Issue by ad0ran** [ninject/Ninject#212]
I have a snippet of code that goes something like this:

`container.Bind<IService>( ).ToMethod(ctx => factory.CreateService(t => ctx.Kernel.Get(t))).When(request => {
    Debug.WriteLine(request.Target.Type.FullName);
    Debug.WriteLine(request.Service.FullName);
    return request.Target.Type.Assembly.FullName.Equals("My.Assembly");
});`

For some reason, the request.Target.Type and the request.Service members are the same thing. So for example, if i'm injecting an IService into say.. MyController, the type of request.Target and request.Service is always equal to IService.

Basically, I want to be able to bind a specific implementation based on the assembly of which the service is being injected into.

**Comment by scott-xu** [comment]
Please try `request.Target.Member.ReflectedType`
